test/services/TimeServiceTest.scala

```diff
@@ -115,11 +111,37 @@ class TimeServiceTest extends ControllerBaseSpec {
         val day = 22
         val expected = new LocalDate(year, month, day)
 
-        val prefix: String = format.substring(0,1)
+        val prefix: String = format.substring(0, 1)
         val date: String = "" + prefix + year + month + day
 
-        service.parseDateWithFormat(date.trim, format.trim) shouldBe expected
+        timeService.parseDateWithFormat(date.trim, format.trim) shouldBe expected
       }
     }
   }
+
+  "getValidYearsApplyMAPreviousYears" should {
+
+    "return empty list if none is passed" in {
+      timeService.getValidYearsApplyMAPreviousYears(None) should have size(0)
+    }
+
+    "return empty list if empty list is passed" in {
+      timeService.getValidYearsApplyMAPreviousYears(None) should have size(0)
+    }
+
+    "return empty list if years < than minim allowed is passed" in {
+      val list = List(models.TaxYear(year = 2002))
+      timeService.getValidYearsApplyMAPreviousYears(Some(list)) should have size(0)
+    }
+
+    "return valid list if years > than minim allowed is passed" in {
+      val year = timeService.getCurrentDate.getYear
+      val from = year - 10
```

Comment:
magic numbers

Reply:
it is not magic it is just random range of year to be able to check a valid range